lib/metasearch_index/deleter.rb

```diff
@@ -0,0 +1,55 @@
+module MetasearchIndex
+  module Deleter
+    class V2
+      class MissingArgument < StandardError; end
+      class NotFound < StandardError; end
+      class UnknownError < StandardError; end
+
+      def initialize(id:, data:)
+        @id = id.presence || raise(MissingArgument, "ID and Type must be supplied.")
+        @data = data.presence || raise(MissingArgument, "Not record provided to insert into Elasticsearch.")
+      end
+
+      def delete
+        processor = ElasticsearchProcessor.new
```

Comment:
Looks like you need to pass in a `MetasearchIndex::client` here (it defaults to the govuk one).